libflux/flux-core/src/semantic/bootstrap.rs

```diff
@@ -35,32 +36,89 @@ pub struct Error {
     pub msg: String,
 }
 
-/// DocPackage represents the documentation for a package and its sub packages
+/// Doc is an enum that can take the form of any form of flux documentation through polymorphism.
 #[derive(Debug, Serialize, Deserialize)]
-pub struct DocPackage {
-    /// the path to the documentation package
-    pub path: String,
-    /// the name of the comments package?
+pub enum Doc {
+    /// Package represents documentation for an entire Flux package.
+    Package(Box<PackageDoc>),
+    /// Value represents documentation for a value exposed from a package.
+    Value(Box<ValueDoc>),
+    /// Builtin represents documentation for a builtin value exposed from a package.
+    Builtin(Box<ValueDoc>),
+    /// Option represents documentation for a option value exposed from a package.
+    Option(Box<ValueDoc>),
```

Comment:
`Option` is a standard type in Rust, will `rustc` interpret this as the Rust `Option` type?

Reply:
I don't think so, and it has not impacted the performance of our tests.

Reply:
From what I am able to tell, and some minor research on rust, I believe in this case the Option keyword is being overloaded, and so does not default to the std::Option type.

Reply:
In order to avoid confusion, I think it may be worthwhile to change our `Option` type to something like `Opt`. At the very least, it would make the code more readable.

Reply:
The change has been made.